### PR TITLE
Try_remove fixes

### PR DESCRIPTION
**Pull request by gulgi**
Used on a "real" project, and bugs were found. And fixed.

Hopefully, I actually understand how the ProjectItem works now. Or better at least. :)
( folder with only folder in has FileCount 0. Just trust the GUID andthings seem to work! )

### CODE REVIEW

File: IncludeToolbox/IncludeFormatter/IncludeLineInfo.cs [hunk 1]
```diff
@@ -92,6 +92,8 @@ public static IncludeLineInfo[] ParseIncludes(string text, bool removeEmptyLines
                 if (multiLineCommentStart > -1 && multiLineCommentStart < includeOccurence) // Multi line comment started in same line, but before #include.
                     continue;
 
+                if (lines[line].Contains(".inl>") || lines[line].Contains(".inl\"")) // Don't remove inline files.
```
Comment (Wumpf) [inline IncludeToolbox/IncludeFormatter/IncludeLineInfo.cs:95]: Hmmm yeah I can see why you're doing this - linker errors between the try & error removal was too aggressive. But this also changes the way sorting is done, so I'd rather make this optional via a flag that is only set for the try&error removal if you don't mind.

**Review by Wumpf — CHANGES_REQUESTED** [review]

File: IncludeToolbox/Commands/TryAndErrorRemoval_Project.cs [hunk 2]
```diff
@@ -114,24 +114,20 @@ static void RecursiveFindFilesInProject(ProjectItems items, ref Queue<ProjectIte
                 var projectItem = item as ProjectItem;
                 if (projectItem == null)
                     continue;
-                if (projectItem.FileCount > 1)
+
+                Guid projectItemKind = new Guid(projectItem.Kind);
+                if (projectItemKind == VSConstants.GUID_ItemType_VirtualFolder ||
```
Comment (Wumpf) [inline IncludeToolbox/Commands/TryAndErrorRemoval_Project.cs:119]: This looks good, thanks a lot! 👍

**Review by Wumpf — APPROVED** [review]

**Comment by Wumpf** [comment]
Ahh I didn't get that GitHub's review function is about the entire pull request and not on a per-commit basis. So if you would just extend the IncludeLineInfo as suggested, I'd love to merge it in and update the official version in the gallery :)

**Comment by Wumpf** [comment]
Uh I hope those GUIDs stayed the same in VS2015 and VS2017. Which one did you try? I'll do a test on the other one then.

File: IncludeToolbox/IncludeFormatter/IncludeLineInfo.cs (hunk 1 above)
Comment (gulgi) [reply IncludeToolbox/IncludeFormatter/IncludeLineInfo.cs:95]: Do you ever want to remove "inline" files?   The ending is problematic, though.
.inc is probably used..?   A list of: "skip these file endings" might be cleanest; yes.

**Comment by gulgi** [comment]
It needs VS2017 to compile, but I'm quite sure I checked the documentation for 2015 as well.

File: IncludeToolbox/IncludeFormatter/IncludeLineInfo.cs (hunk 1 above)
Comment (Wumpf) [reply IncludeToolbox/IncludeFormatter/IncludeLineInfo.cs:95]: No I don't think so, but the IncludeLineInfo is also used for other tools, namely sorting (which I do use quite a lot). The sorting however, can easily be configured by the user to account for inl files.

You have a good point for the .inc though! I think it is best if the user could configure a list of regexes for "never remove this" that we can check before we try to remove an include. I'll get right on it and put .inc, .inl and _inl.h on the list by default.

Comment (Wumpf) [reply IncludeToolbox/IncludeFormatter/IncludeLineInfo.cs:95]: Halfway in, I just realized that this feature is already there. Woops.
Going to put a meaningful default in there and test it again.
Options>Include Toolbox>Try and Error Include Removal

**Comment by Wumpf** [comment]
I've cherry picked the project "try and error removal" fixes and put meaningful default values to the TryAndError ignore list.

**Pull request closed without merging**